packages/vhd-lib/createStreamNbd.js

```diff
@@ -69,10 +79,36 @@ exports.createNbdVhdStream = async function createVhdStream(nbdClient, sourceStr
     }
   }
 
+  const totalLength = (offsetSector + blockSizeInSectors + 1) /* end footer */ * SECTOR_SIZE
+
+  let lengthRead = 0
+  let lastUpdate = 0
+  let lastLengthRead = 0
+
+  function throttleEmitProgress() {
+    const now = Date.now()
+
+    if (
+      lengthRead - lastLengthRead > (minTresholdPercentBetweenProgressEmit / 100) * totalLength ||
+      (now - lastUpdate > maxDurationBetweenProgressEmit && lengthRead !== lastLengthRead)
+    ) {
+      stream.emit('progress', lengthRead / totalLength)
+      lastUpdate = now
+      lastLengthRead = lengthRead
+    }
+  }
+
+  const interval = setInterval(throttleEmitProgress, maxDurationBetweenProgressEmit)
+  function* trackAndYield(buffer) {
+    lengthRead += buffer.length
+    throttleEmitProgress()
+    yield buffer
+  }
```

Comment:
Why are you using a generator here, it only yields one value?

Reply:
There is no specific reason. A simple function that returns the buffer is more appropriate?

Reply:
Yep, keep think as simple as possible.

It will likely helps both maintenance and performance.